packages/deployer/src/compiler.ts

```diff
@@ -207,6 +237,36 @@ export class Compiler {
         const sourceMapRuntime = compiled.contracts[contractIdentifier].srcmapRuntime;
         const sources = _.keys(compiled.sources);
         const updated_at = Date.now();
+
+        // There is no function overloading in typescript, so we must change the Typescript ABI interface.
```

Comment:
There are some options to overloading in TS. The documentation gives a simple example:

http://www.typescriptlang.org/docs/handbook/functions.html#overloads

I don't know TS well enough to tell if this is to be preferred over the current solution. It seems like you'd have to manually do the type-pattern-matching, but that's doable.

@fabioberger @LogvinovLeon what's your take on this?

Reply:
I didn't actually realize we owned the code that produces Typescript from the artifacts. This could be interesting.